File: mr_patent_backend/src/main/java/com/d208/mr_patent_backend/domain/voca/dto/level/LevelDTO.java
package com.d208.mr_patent_backend.domain.voca.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 레벨 정보를 담는 DTO
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LevelDTO {
    private Byte level_id;
    private String level_name;
    private boolean is_accessible;
    private Byte best_score;
    private boolean is_passed;
}

File: mr_patent_android/app/src/main/java/com/ssafy/mr_patent_android/data/model/dto/LevelDto.kt
package com.ssafy.mr_patent_android.data.model.dto

import com.google.gson.annotations.SerializedName

data class LevelDto(
    @SerializedName("levels")val levels: List<Level>
){
    data class Level(
        @SerializedName("best_score")val best_score: Int,
        @SerializedName("accessible")val is_accessible: Boolean,
        @SerializedName("passed")val is_passed: Boolean,
        @SerializedName("level_id")val level_id: Int,
        @SerializedName("level_name")val level_name: String,
        @SerializedName("count")val count: Int=-1,
    )
}